### DISCUSSION TITLE
Cara membuat ascending & descending di shiny

### DISCUSSION THREAD

**Discussion by vlvivian** [teamalgoritma/community#106] [Q&A] [answered]
Halo Pak/Bu, ijin bertanya apa yang harus saya masukkan ke dalam 'arrange' dan 'choice' di radio button supaya dapat connect ke plot yang muncul ? 

Code saya sebagai berikut:

- Server :
bank_dm <- bank %>% 
      filter(job == input$input_dm) %>% 
      group_by(day,month) %>% 
      summarise(mean_dm=mean(duration)) %>% 
      ungroup() %>% 
      arrange(-mean_dm)

- ui
radioButtons(
       inputId = "sort",
       label = "Sorting",
       choices = c("Ascending", "Descending")
     )

Terima kasih

**Comment by inytss** [comment, marked as answer]
Hi @vlvivian ,
Untuk membuat sebuah input ascending dan descending, dalam server Ibu perlu membuat sebuah ifelse condition sesuai dengan kondisi yang Ibu berikan pada choices. Berikut contoh pembuatan ifelse condition pada server.
```
output$plot <- renderHighchart({
    if (input$option == "death") {
      ts_deaths_long %>%
        mutate(datetime = ymd(datetime)) %>%
        filter(country %in% input$country) %>%
        group_by(country, datetime) %>%
        summarise(deaths = sum(deaths)) %>%
        filter(datetime >= input$dateSelector[1] &
                 datetime < input$dateSelector[2]) %>%
        
        hchart(.,
               type = "line",
               hcaes(x = datetime,
                     y = deaths,
                     group = country)) %>%
        hc_title(text = "The Total Number of Deaths") %>%
        hc_subtitle(text = "The data is compiled by the Johns Hopkins University Center") %>%
        hc_plotOptions(line = list(
          lineWidth = 4,
          allowPointSelect = TRUE,
          marker = list(
            enabled = FALSE,
            radius = 1,
            symbol = "circle"
          )
        )) %>%
        hc_add_theme(hc_theme_538()) %>%
        hc_exporting(enabled = TRUE)
    }
    
    else {
      ts_confirmed_long %>%
        mutate(datetime = ymd(datetime)) %>%
        filter(country %in% input$country) %>%
        group_by(country, datetime) %>%
        summarise(confirmed = sum(confirmed)) %>%
        filter(datetime >= input$dateSelector[1] &
                 datetime < input$dateSelector[2]) %>%
        hchart(.,
               type = "line",
               hcaes(x = datetime,
                     y = confirmed,
                     group = country)) %>%
        hc_title(text = "The Total Number of Confirmed Cases") %>%
        hc_subtitle(text = "The data is compiled by the Johns Hopkins University Center ") %>%
        hc_plotOptions(line = list(
          lineWidth = 4,
          allowPointSelect = TRUE,
          marker = list(
            enabled = FALSE,
            radius = 1,
            symbol = "circle"
          )
        )) %>%
        hc_add_theme(hc_theme_538()) %>%
        hc_exporting(enabled = TRUE)
      
    }
    
  })
```

Pada contoh code yang saya berikan, saya memiliki sebuah input dengan nama `options` dengan pilihan "death" dan "confirm". Karena hanya terdapat 2 kondisi saja, jadi saya coba buat untuk kondisi "death" akan menjadi sebuah visualisasi dengan case "deacth" sedangkan yang lainnya akan langsung menjadi "confirm" case.

Semoga membantu :)

**Comment by NabiilahArdini** [comment]
Halo @vlvivian ,

Berikut contoh lain yang mungkin lebih singkat terkait pertanyaan Anda. Anda perlu membuat skema `ifelse()` karena sayangnya kebutuhan Anda berkaitan dengan penggunaan fungsi R (arrange & desc) yang berbeda antar kondisinya. 

1. Fungsi pada radioButton Anda sudah tepat. 
2. Tambahkan kondisi ifelse yang lebih detail dengan struktur `if (kondisi A) {output A} else {output B}` setelah preparasi data Anda, yaitu ketika ingin melakuan print data.

Berkut contohnya, ketika saya ingin menampilkan data iris secara arrange ascending/descending (perhatikan bagian server):

```
library(shiny)
library(dplyr)

ui <- fluidPage(
  
  titlePanel("Example Data"),
  sidebarLayout(
    sidebarPanel(selectInput("species", "Pick Species", unique(iris$Species)),
                 radioButtons(
                   inputId = "sort",
                   label = "Sorting",
                   choices = c("Ascending", "Descending")
                 )), 
    mainPanel(tableOutput("table"))
    )
  
)

server <- function(input, output, session) {
  
  output$table <- renderTable({
    
    data <- iris %>% 
      filter(Species == input$species) %>%
      select(Species, Petal.Length)
    
    # pilihan cara menampilkan tabel
    if(input$sort == "Ascending") 
      { data %>% arrange(Petal.Length) } else
      { data %>% arrange(desc(Petal.Length)) }
    
  })
  
}

shinyApp(ui, server)
```

Kemungkinan besar untuk penerapannya di code Anda seperti ini di bagian server:
```
bank_dm <- bank %>%
filter(job == input$input_dm) %>%
group_by(day,month) %>%
summarise(mean_dm=mean(duration)) %>%
ungroup()

if(input$sort == "Ascending") 
      { bank_dm %>% arrange(mean_dm) } else
      { bank_dm %>% arrange(desc(mean_dm)) }
```

Selain itu, bila Anda tertarik untuk langsung mengaplikasikan fitur ascending/descending pada tampilan table di shiny, Anda dapat menggunakan fitur Data Tables dari Shiny. Berikut dokumentasinya: https://shiny.rstudio.com/articles/datatables.html

Semoga dapat informasi ini dapat membantu :)
